domain/src/main/java/life/qbic/domain/user/EncryptedPassword.java

```diff
@@ -0,0 +1,110 @@
+package life.qbic.domain.user;
+
+import java.io.Serial;
+import java.util.Objects;
+import life.qbic.apps.datamanager.ApplicationException;
+import life.qbic.domain.usermanagement.policies.PasswordEncryptionPolicy;
+import life.qbic.domain.usermanagement.policies.PasswordPolicy;
+import life.qbic.domain.usermanagement.policies.PolicyCheckReport;
+import life.qbic.domain.usermanagement.policies.PolicyStatus;
+
+/**
+ * <b>Encrypted Password</b>
+ * <p>
+ * Represents an encrypted user password.
+ *
+ * @since 1.0.0
+ */
+public class EncryptedPassword {
+
+  private String encryptedPassword;
+
+  /**
+   * Sets a password for the current user.
+   *
+   * <p>Beware that the password gets validated against the current password policy. If the
+   * password violates the policy, an {@link PasswordValidationException} is thrown.
+   *
+   * <p>The password is then stored in an encrypted form, controlled by the {@link
+   * PasswordEncryptionPolicy}.
+   *
+   * <p>It is the client's responsibility to reset the raw password, after it has been set for the
+   * user and encrypted.
+   *
+   * @param rawPassword the new user password
+   * @throws PasswordValidationException if the user password is too weak
+   * @since 1.0.0
+   */
+  public static EncryptedPassword from(char[] rawPassword) throws PasswordValidationException {
+    PolicyCheckReport policyCheckReport = PasswordPolicy.create().validate(rawPassword);
+    if (policyCheckReport.status() == PolicyStatus.FAILED) {
+      throw new PasswordValidationException(policyCheckReport.reason());
+    }
+    var password = new EncryptedPassword();
+    password.setEncryptedPassword(encryptPassword(rawPassword));
+    return password;
+  }
+
+  protected static EncryptedPassword fromEncrypted(String encryptedPassword) {
+    return new EncryptedPassword(encryptedPassword);
+  }
+
+  protected EncryptedPassword(String encryptedPassword) {
+    this.encryptedPassword = encryptedPassword;
+  }
+
+  private EncryptedPassword() {
+    super();
+  }
+
+  private void setEncryptedPassword(String encryptedPassword) {
+    this.encryptedPassword = encryptedPassword;
+  }
```

Comment:
why do you need this method?

Reply:
no, it should not. JPA support and it is effectively final